x/pki/client/cli/tx_assign_vid.go

```diff
@@ -0,0 +1,48 @@
+package cli
+
+import (
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/pki/types"
+)
+
+var _ = strconv.Itoa(0)
+
+func CmdAssignVid() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "assign-vid [subject] [subject-key-id] [vid]",
+		Short: "Broadcast message AssignVid",
+		Args:  cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			subject := viper.GetString(FlagSubject)
+			subjectKeyID := viper.GetString(FlagSubjectKeyID)
+			vid := viper.GetInt32(FlagVid)
+
+			msg := types.NewMsgAssignVid(
+				clientCtx.GetFromAddress().String(),
+				subject,
+				subjectKeyID,
+				vid,
+			)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
```

Comment:
Please use the following code instead of the last lines(see other commands):
```
                        err = tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
			if cli.IsWriteInsteadReadRPCError(err) {
				return clientCtx.PrintString(cli.LightClientProxyForWriteRequests)
			}
                        return err

```

Reply:
ok